server/src/main/java/org/candlepin/async/tasks/HealEntireOrgJob.java

```diff
@@ -72,18 +71,23 @@ public HealEntireOrgJob(Entitler entitler, ConsumerCurator consumerCurator, Owne
     public Object execute(JobExecutionContext context) throws JobExecutionException {
         try {
 
-            final JobArguments arguments = context.getJobArguments();
-            final String ownerKey = arguments.getAsString(OWNER_KEY);
+            JobArguments arguments = context.getJobArguments();
+            String ownerKey = arguments.getAsString(OWNER_KEY);
             Owner owner = ownerCurator.getByKey(ownerKey);
+            if (owner == null) {
+                log.warn("Healing attempted against non-existent org id \"{}\"", ownerKey);
+                throw new JobExecutionException(
+                    i18n.tr("Healing attempted against non-existent org id \"{}\"", ownerKey));
+            }
             if (owner.isAutobindDisabled() || owner.isContentAccessEnabled()) {
                 String caMessage = owner.isContentAccessEnabled() ?
                     " because of the content access mode setting" : "";
-                throw new BadRequestException(
+                throw new JobExecutionException(
```

Comment:
Remember to set the terminal flag to true here: 

```new JobExecutionException(i18n.tr("..."), true);```

Reply:
Done. I misunderstood your earlier comment.